core/platform/datastore/gae_datastore_services.py

```diff
@@ -19,9 +19,34 @@
 from __future__ import absolute_import  # pylint: disable=import-only-modules
 from __future__ import unicode_literals  # pylint: disable=import-only-modules
 
+from google.appengine.datastore import datastore_query
 from google.appengine.ext import ndb
 
 
+def make_cursor(urlsafe=None):
+    """Returns a database cursor that represents a relative position in a query.
+
+    The position denoted by a Cursor is relative to a result in a query even
+    if the result has been removed from the given query. Usually to position
+    immediately after the last result returned by a batch.
+
+    A cursor should only be used on a query with an identical signature to the
+    one that produced it or on a query with its sort order reversed.
+
+    A Cursor constructed with no arguments points the first result of any
+    query. If such a Cursor is used as an end_cursor no results will ever be
+    returned.
+
+    Args:
+        urlsafe: str. A base64-encoded serialization of a cursor. If None, the
```

Comment:
str|None

Should we call it urlsafe_cursor?

Reply:
Done.

I don't think so, I lifted all this documentation straight out of the real implementation, so I think we should stay faithful to it for the sake of consistency.

Reply:
Hm -- this is now code that we own, instead of third-party code, so I lean towards urlsafe_cursor or urlsafe_serialized_cursor because it more clearly describes the thing that's being passed in (for one thing, there's actually a noun there).

However, this doesn't block approval -- it's your decision! Just stating my reasoning.

Reply:
Yeah point well taken, done.